### ISSUE TITLE
ELMS help agent

### ISSUE THREAD

**Issue by btopro** [elmsln/elmsln#1296]
- [ ] Instructional Design checklist via ChecklistAPI #1198
- [ ] 'terminal'-esk / voice based system to ask questions / for help and get recommended actions
- related past request Text and Voice Commands and on-demand interfaces #958

something like #968 #567 but visualizing / workflow came to me this morning.

Using http://materializecss.com/modals.html bottom sheet modals we could have the current "jarvis" prototype fly up from the bottom. It could also use a glowing svg elmsln network icon set on top of a css hexagon that's juuuuuuust off the bottom of the screen (perhaps barely visible on certain screen sizes,. dono). When elms is talking, the icon is glowing / pulsing, when listening it's light blue (or something like that).

This could also be used for a way of accessing #502 or being them in general. The elms help agent could also do things like search wikipedia for a highlighted term (something chrome mobile has a quick option for), which could then trigger a Side-Nav from the right as "related" content kinda deal.

This also could solve the "what do I do with this" problem when you first install. Could set a flag for "new_system" and when new system is triggered on 1st page load of online.elmsln.local (as unauthenticated user). The help agent could slide up automatically and start talking / set focus there (a11y) and then start welcoming the user to the system and asking them to login, focusing on the login prompt after finishing saying it.

This isn't trivial at all, but a first step could be styling the bottom modal to take the place of the current Jarvis prototype text area. This way it at least looks theme cohesive. Then making the Shapes API to do the CSS hexagons with ease. Then making a smaller elmsln svg to glow with css at the right animation rate / states.

**Comment by btopro** [comment]
#1198

**Comment by btopro** [comment]
Just like it says. We could use ChecklistAPI to create a list of common instructional design things that should be checked when building a course out and also we could suggest things based on analysis of the course. Like

4 of your 100 pages use the Instructional Significance setting, would you like to add more?
You have 3 quizzes listed
There is currently 22 hours of reading
There are 40 videos totalling approximately 12 hours.
This is both feedback as well as statistics which could be used as a HUD of sorts for making one think about and improve the design of a course. might plug in nicely with the #842 if it's role centric so staff get additional options
